File: lib/models/circuit.dart
import 'package:logic_gates/models/input.dart';
import 'package:logic_gates/models/output.dart';

class Circuit {
  Circuit({
    List<Input>? inputs,
    List<Output>? outputs,
  })  : _inputs = inputs ?? [],
        _outputs = outputs ?? [];

  // TODO: Give inputs a purpose
  late final List<Input> _inputs;
  late final List<Output> _outputs;

  List<bool> evaluate() {
    if (_outputs.isEmpty) {
      return [];
    }

    return _outputs.map((e) => e.evaluate()).toList();
  }
}

File: pkg/models/circuit.go
package models

type Circuit struct {
	Inputs  []*Input
	Outputs []*Output
}

func (c *Circuit) AddInput(input *Input) {
	c.Inputs = append(c.Inputs, input)
}

func (c *Circuit) AddOutput(output *Output) {
	c.Outputs = append(c.Outputs, output)
}

func (c *Circuit) Evaluate() bool {
	return c.Inputs[0].Evaluate()
}
